### PR TITLE
Remove unwraps from non-test code in queue/redis

### PR DESCRIPTION
**Pull request by svix-dylan**
We had a lot of unwraps in our Redis-based queue implementation. This PR refactors them to be `.expect()`s where appropriate, or to bubble errors up to a calling function that catches them.

### CODE REVIEW

**Comment by svix-dylan** [comment]
Would love feedback on this. I've only done a cursory run through. Open questions include, in functions that call and catch errors from the `migrate_` functions -- how do we want to handle these errors? Currently I'm just passing them to a `trace::error!`. But they are nonrecoverable, right?

File: server/svix-server/src/queue/redis.rs [hunk 1]
```diff
@@ -968,13 +970,18 @@ pub mod tests {
             }
 
             // v1 to v2
-            migrate_list(&mut conn, v1_main, v2_main).await;
-            migrate_list(&mut conn, v1_processing, v2_processing).await;
-            migrate_sset(&mut conn, v1_delayed, v2_delayed).await;
+            let r1 = migrate_list(&mut conn, v1_main, v2_main).await;
```
Comment (tasn) [inline server/svix-server/src/queue/redis.rs:973]: The way to do it in tests is just to unwrap. This is too cumbersome.

File: server/svix-server/src/queue/redis.rs [hunk 2]
```diff
@@ -367,15 +370,17 @@ fn to_redis_key(delivery: &TaskQueueDelivery) -> String {
     format!(
```
[the remark this answers is not in the document]
Comment (tasn) [reply server/svix-server/src/queue/redis.rs:370]: I agree. I think this is much more concerning than e.g. the migrations.

**Comment by tasn** [comment]
This was now fixed by @svix-james in another PR, no?

**Comment by svix-jplatte** [comment]
Outdated in many ways.

**Pull request closed without merging**